tensorboard/uploader/uploader.py

```diff
@@ -295,23 +304,32 @@ def send_requests(self, run_to_events):
             # If later events arrive with a mismatching plugin_name, they are
             # ignored with a warning.
             metadata = self._tag_metadata.get(time_series_key)
+            first_in_time_series = False
```

Comment:
This mechanism of detecting 'first time' is a little entangled hard to follow.  How about maintaining a collection of `plugin_names_seen` and testing `if plugin_name in plugin_names_seen:`?

Related, the 'first time' gets reset for every `_upload_once()`.  Is that desirable, or should we maintain a singleton `plugin_names_seen` (or similar) that persists over cycles?  In this case you'd change the error message too, of course, so it's not time-series-specific but just appears once per plugin.